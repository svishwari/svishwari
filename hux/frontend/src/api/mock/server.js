import { belongsTo, createServer, Factory, hasMany, Model } from "miragejs"
import config from "@/config"
import AppSerializer from "./serializer"

// routes
import { defineRoutes } from "./routes"

// seeds
import seeds from "./seeds"

// factories
import { audience as audienceFactory } from "./factories/audiences"
import { customer, customerProfile } from "./factories/customers"
import dataSourceFactory from "./factories/dataSource"
import { destination as destinationFactory } from "./factories/destination"
import { engagement as engagementFactory } from "./factories/engagement"
import modelFactory from "./factories/model"
import audiencePerformanceFactory from "./factories/audiencePerformance"
import dataExtensionFactory from "./factories/dataExtensions"
import deliveryFactory from "./factories/delivery"
import campaignFactory from "./factories/campaigns"
import { notification as notificationFactory } from "./factories/notifications"

export function makeServer({ environment = "development" } = {}) {
  // models
  const models = {
    audience: Model.extend({
      destinations: hasMany("destination"),
      engagements: hasMany("engagement"),
    }),
    customer: Model,
    customerProfile: Model,
    dataSource: Model,
    destination: Model.extend({
      destinationable: belongsTo({ polymorphic: true }),
    }),
    engagement: Model.extend(),
    model: Model,
    audiencePerformance: Model,
    dataExtension: Model,
    deliveryFactory: Model,
    campaign: Model,
    notification: Model,
  }

  const factories = {
    audience: Factory.extend(audienceFactory),
    customer: Factory.extend(customer),
    customerProfile: Factory.extend(customerProfile),
    dataSource: Factory.extend(dataSourceFactory),
    destination: Factory.extend(destinationFactory),
    engagement: Factory.extend(engagementFactory),
    model: Factory.extend(modelFactory),
    audiencePerformance: Factory.extend(audiencePerformanceFactory),
    dataExtension: Factory.extend(dataExtensionFactory),
    delivery: Factory.extend(deliveryFactory),
    campaign: Factory.extend(campaignFactory),
    notification: Factory.extend(notificationFactory),
  }

  const server = createServer({
    environment: environment,
    models: models,
    factories: factories,
    seeds: seeds,
    routes() {
      this.urlPrefix = config.apiUrl
      this.namespace = config.apiBasePath
      this.timing = 1000
      defineRoutes(this)

      // pass requests to external APIs through
      this.passthrough(`${config.oktaUrl}/**`)
    },
    serializers: {
      application: AppSerializer,
      audience: AppSerializer.extend({
        include: ["destinations", "engagements"],
      }),
      customer: AppSerializer.extend({
        root: true,
      }),
    },
  })

  return server
}
