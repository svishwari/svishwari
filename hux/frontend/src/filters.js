/**
 * Globally registerd Vue filters
 */
import moment from "moment"

export default {
  /**
   * Formats a datetime field to human friendly date.
   *
   * @param {*} value Datetime eg. 2021-03-17T23:29:49.351Z
   * @param {*} format Format eg. "MM/D/YYYY [at] hh:ss A"
   * @return {string} Formatted date time string eg. 17/3/2021 at 11:29pm
   */
  Date(value, format = "MM/D/YYYY [at] hh:ss A") {
    if (!value) return ""

    if (format === "relative") return moment(value).fromNow()

    if (format === "calendar") return moment(value).calendar()

    return moment(value).format(format)
  },

  /**
   * Formats any empty data field with a placeholder.
   *
   * @param {*} value Empty string eg. ""
   * @param {*} placeholder Placeholder eg. "N/A"
   * @returns Formatted empty data field eg. "N/A"
   */
  Empty(value, placeholder = "â€”") {
    if (!value) return placeholder
    return value
  },

  Numeric(value, round = false, abbreviate = false, append = "") {
    if (isNaN(value)) return ""

    let abrv = ""

    if (abbreviate) {
      if (value >= 1000000) {
        value = value / 1000000
        abrv = "m"
      } else if (value >= 1000) {
        value = value / 1000
        abrv = "k"
      }
    }

    return (
      value.toLocaleString("en-US", {
        minimumFractionDigits: 0,
        maximumFractionDigits: round && Number(value) ? 0 : 2,
      }) +
      abrv +
      append
    )
  },

  /**
   * Formats a string with title case.
   *
   * @param {*} value The string eg. "active Customers"
   * @returns Title cased string eg. "Active Customers"
   */
  TitleCase(value) {
    return value
      .replace(/([A-Z])/g, (match) => ` ${match}`)
      .replace(/^./, (match) => match.toUpperCase())
      .trim()
  },
  /**
   * Formats any string(fullname) to shortname.
   *
   * @param {*} value The string eg. "John petro"
   * @returns shortname string eg. "JP"
   */
  shortName(value) {
    return value
      .split(" ")
      .map((n) => n[0])
      .join("")
  },
  /**
   * Format the number with comma
   *
   * @param {*} value the number eg "125767"
   * @returns formatted number
   */
  FormatSize(value) {
    return Number(value).toLocaleString()
  },
}
