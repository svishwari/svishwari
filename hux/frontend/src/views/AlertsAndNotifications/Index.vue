<template>
  <div class="notification-wrap">
    <page-header :header-height-changes="'py-3'">
      <template slot="left">
        <breadcrumb :items="breadcrumbItems" />
      </template>
    </page-header>
    <page-header class="top-bar mb-3" :header-height="71">
      <template #left>
        <v-icon medium color="black lighten-3">mdi-filter-variant</v-icon>
        <v-icon medium color="black lighten-3" class="pl-4">mdi-magnify</v-icon>
      </template>

      <template #right>
        <huxButton
          variant="primary base"
          icon="keyboard-return"
          is-custom-icon
          class="ma-2 caption no-shadow mr-0"
          size="large"
          is-tile
          height="40"
          icon-size="18"
          icon-color="white"
          icon-variant="base"
          data-e2e="notification-return"
          @click="goBack()"
        >
          Return to previous page
        </huxButton>
      </template>
    </page-header>
    <v-progress-linear :active="loading" :indeterminate="loading" />
    <v-row v-if="!loading" class="pb-7 pl-3 white">
      <hux-data-table
        :columns="columnDefs"
        :data-items="notifications"
        sort-column="created"
        sort-desc
      >
        <template #row-item="{ item }">
          <td
            v-for="header in columnDefs"
            :key="header.value"
            :class="{
              'fixed-column': header.fixed,
              'v-data-table__divider': header.fixed,
              'primary--text': header.fixed,
            }"
            class="col-overflow"
            :style="{ width: header.width, left: 0 }"
          >
            <div v-if="header.value == 'created'">
              <time-stamp :value="item['created']" />
            </div>
            <div v-if="header.value == 'notification_type'">
              <status
                :status="item['notification_type']"
                :show-label="true"
                :icon-size="17"
              />
            </div>
            <tooltip v-if="header.value == 'description'" position-top>
              <template #label-content>
                <span>{{ item[header.value] }}</span>
              </template>
              <template #hover-content>
                {{ item[header.value] }}
              </template>
            </tooltip>
          </td>
        </template>
      </hux-data-table>
    </v-row>
    <v-divider v-if="enableLazyLoad" class="hr-devider"></v-divider>
    <v-progress-linear v-if="enableLazyLoad" active indeterminate />
    <observer v-if="notifications.length" @intersect="intersected"></observer>
  </div>
</template>

<script>
import { mapActions, mapGetters } from "vuex"
import PageHeader from "@/components/PageHeader"
import Breadcrumb from "@/components/common/Breadcrumb"
import huxButton from "@/components/common/huxButton"
import HuxDataTable from "../../components/common/dataTable/HuxDataTable.vue"
import TimeStamp from "../../components/common/huxTable/TimeStamp.vue"
import Status from "../../components/common/Status.vue"
import Tooltip from "@/components/common/Tooltip.vue"
import Observer from "@/components/common/Observer"

export default {
  name: "AlertsAndNotifications",
  components: {
    PageHeader,
    Breadcrumb,
    huxButton,
    HuxDataTable,
    TimeStamp,
    Status,
    Tooltip,
    Observer,
  },
  data() {
    return {
      breadcrumbItems: [
        {
          text: "Alerts & Notifications",
          disabled: true,
          icon: "bell",
        },
      ],
      categoryItems: [
        { name: "Orchestration" },
        { name: "Decisioning" },
        { name: "Data management" },
      ],
      alertTypeItems: [
        { name: "Success", modelIcon: "Success" },
        { name: "Critical", modelIcon: "Critical" },
        { name: "Feedback", modelIcon: "Feedback" },
        { name: "Informational", modelIcon: "Informational" },
      ],
      columnDefs: [
        {
          text: "Time",
          value: "created",
          width: "180px",
        },
        {
          text: "Type",
          value: "notification_type",
          width: "180px",
        },
        {
          text: "Description",
          value: "description",
          width: "auto",
        },
      ],
      sortColumn: "created",
      sortDesc: true,
      loading: false,
      enableLazyLoad: false,
      lastBatch: 0,
      batchDetails: {
        batchSize: 25,
        batchNumber: 1,
        isLazyLoad: false,
      },
    }
  },
  computed: {
    ...mapGetters({
      notifications: "notifications/list",
      totalNotifications: "notifications/total",
    }),
  },

  async mounted() {
    this.loading = true
    await this.fetchNotificationsByBatch()
    this.calculateLastBatch()
    this.loading = false
    this.enableLazyLoad = true
    if (this.notifications.length === 0) {
      this.enableLazyLoad = false
    }
  },
  methods: {
    ...mapActions({
      getAllNotifications: "notifications/getAll",
    }),
    goBack() {
      this.$router.go(-1)
    },
    intersected() {
      if (this.batchDetails.batchNumber <= this.lastBatch) {
        this.batchDetails.isLazyLoad = true
        this.fetchNotificationsByBatch()
      } else {
        this.enableLazyLoad = false
      }
    },
    async fetchNotificationsByBatch() {
      await this.getAllNotifications(this.batchDetails)
      this.batchDetails.batchNumber++
    },
    calculateLastBatch() {
      this.lastBatch = Math.ceil(
        this.totalNotifications / this.batchDetails.batchSize
      )
    },
  },
}
</script>
<style lang="scss" scoped>
.notification-wrap {
  background: white;
  ::v-deep .menu-cell-wrapper .action-icon {
    display: none;
  }
  .top-bar {
    margin-top: 1px;
    .v-icon--disabled {
      color: var(--v-black-lighten3) !important;
      font-size: 24px;
    }
    .text--refresh {
      margin-right: 10px;
    }
  }
  .hux-data-table {
    margin-top: 1px;
    table {
      tr {
        td {
          font-size: 14px;
          height: 63px;
        }
      }
    }
  }
  ::v-deep .menu-cell-wrapper :hover .action-icon {
    display: initial;
  }
  .icon-border {
    cursor: default !important;
  }
}
.radio-div {
  margin-top: -11px !important;
}

.backGround-header-dropdown {
  background-color: var(--v-primary-lighten2) !important;
}
::v-deep .hux-dropdown {
  .v-btn__content {
    color: var(--v-darkBlue-base) !important;
  }
}
.col-overflow {
  white-space: nowrap !important;
  overflow: hidden !important;
  text-overflow: ellipsis !important;
}

::v-deep
  .theme--light.v-data-table
  > .v-data-table__wrapper
  > table
  > tbody
  > tr:hover:not(.v-data-table__expanded__content):not(.v-data-table__empty-wrapper) {
  background: white !important;
}
::v-deep
  .theme--light.v-data-table
  > .v-data-table__wrapper
  > table
  > thead
  > tr:last-child
  > th {
  padding-left: 32px !important;
}
::v-deep
  .theme--light.v-data-table
  > .v-data-table__wrapper
  > table
  > tbody
  > tr:not(:last-child)
  > td:not(.v-data-table__mobile-row),
.theme--light.v-data-table
  > .v-data-table__wrapper
  > table
  > tbody
  > tr:not(:last-child)
  > th:not(.v-data-table__mobile-row) {
  padding-left: 32px !important;
}
::v-deep .v-data-table > .v-data-table__wrapper > table > tbody > tr > td,
.v-data-table > .v-data-table__wrapper > table > thead > tr > td,
.v-data-table > .v-data-table__wrapper > table > tfoot > tr > td {
  padding-left: 32px !important;
}
.hr-devider {
  margin-top: -27px !important;
}
</style>
