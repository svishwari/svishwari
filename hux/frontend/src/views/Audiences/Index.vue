<template>
  <div class="audiences-wrap white">
    <page-header :header-height-changes="'py-3'">
      <template slot="left">
        <breadcrumb :items="breadcrumbItems" />
      </template>
      <template slot="right">
        <v-icon size="22" color="black lighten-3" class="icon-border pa-2 ma-1"
          >mdi-download</v-icon
        >
      </template>
    </page-header>
    <page-header class="top-bar" :header-height="71">
      <template slot="left">
        <v-icon medium color="black lighten-3">mdi-filter-variant</v-icon>
        <v-icon medium color="black lighten-3" class="pl-6">mdi-magnify</v-icon>
      </template>

      <template slot="right">
        <v-icon medium color="black lighten-3 refresh">mdi-refresh</v-icon>
        <router-link
          :to="{ name: 'AudienceConfiguration' }"
          class="text-decoration-none"
          append
        >
          <huxButton
            icon="mdi-plus"
            icon-position="left"
            variant="primary"
            size="large"
            is-tile
            class="ma-2 font-weight-regular no-shadow mr-0"
          >
            Audience
          </huxButton>
        </router-link>
      </template>
    </page-header>
    <v-progress-linear :active="loading" :indeterminate="loading" />
    <v-row v-if="!loading" class="pt-3 pb-7 pl-3 white">
      <hux-data-table
        v-if="isDataExists"
        :columns="columnDefs"
        :data-items="audienceList"
      >
        <template #row-item="{ item }">
          <td
            v-for="header in columnDefs"
            :key="header.value"
            :class="{
              'fixed-column': header.fixed,
              'v-data-table__divider': header.fixed,
              'primary--text': header.fixed,
            }"
            :style="{ width: header.width, left: 0 }"
          >
            <div v-if="header.value == 'name'" class="w-100 d-flex">
              <span v-if="item.is_lookalike == true" class="mr-3">
                <tooltip>
                  <template #label-content>
                    <icon
                      type="lookalike"
                      :size="20"
                      color="neroBlack"
                      class="mr-2"
                    />
                  </template>
                  <template #hover-content>Lookalike audience</template>
                </tooltip>
              </span>
              <menu-cell
                :value="item[header.value]"
                :menu-options="getActionItems(item)"
                route-name="AudienceInsight"
                :route-param="item['id']"
              />
            </div>
            <div v-if="header.value == 'status'" class="text-caption">
              <status
                :status="item[header.value]"
                :show-label="true"
                class="d-flex"
                :icon-size="17"
              />
            </div>
            <div v-if="header.value == 'size'">
              <size :value="item[header.value]" />
            </div>
            <div v-if="header.value == 'destinations'">
              <div
                v-if="item[header.value] && item[header.value].length > 0"
                class="d-flex align-center"
              >
                <div class="d-flex align-center">
                  <tooltip
                    v-for="destination in item[header.value]"
                    :key="`${item.id}-${destination.type}`"
                  >
                    <template #label-content>
                      <logo
                        :key="destination.id"
                        class="mr-1"
                        :type="destination.type"
                        :size="18"
                      />
                    </template>
                    <template #hover-content>
                      <span>{{ destination.name }}</span>
                    </template>
                  </tooltip>
                </div>
                <span
                  v-if="item[header.value] && item[header.value].length > 3"
                  class="ml-1"
                >
                  + {{ item[header.value].length - 2 }}
                </span>
              </div>
              <span v-else>—</span>
            </div>
            <div v-if="header.value == 'last_delivered'">
              <time-stamp :value="item[header.value]" />
            </div>
            <div
              v-if="
                header.value == 'update_time' || header.value == 'create_time'
              "
            >
              <time-stamp :value="item[header.value]" />
            </div>
            <div
              v-if="
                header.value == 'updated_by' || header.value == 'created_by'
              "
            >
              <avatar :name="item[header.value]" />
            </div>
          </td>
        </template>
      </hux-data-table>

      <empty-page v-if="!isDataExists">
        <template #icon>mdi-alert-circle-outline</template>
        <template #title>Oops! There’s nothing here yet</template>
        <template #subtitle>
          You currently have no audiences created! You can create the
          <br />framework first then complete the details later. <br />Begin by
          selecting the button below.
        </template>
        <template #button>
          <router-link
            :to="{ name: 'AudienceConfiguration' }"
            class="route-link text-decoration-none"
            append
          >
            <huxButton
              icon="mdi-plus"
              icon-position="left"
              variant="primary"
              size="large"
              is-tile
              class="ma-2 font-weight-regular"
            >
              Audience
            </huxButton>
          </router-link>
        </template>
      </empty-page>
    </v-row>

    <look-alike-audience
      :toggle="showLookAlikeDrawer"
      :selected-audience="selectedAudience"
      @onToggle="(val) => (showLookAlikeDrawer = val)"
      @onError="onError($event)"
    />

    <hux-alert
      v-model="flashAlert"
      :type="alert.type"
      :message="alert.message"
    />
  </div>
</template>

<script>
import { mapGetters, mapActions } from "vuex"

import PageHeader from "@/components/PageHeader"
import EmptyPage from "@/components/common/EmptyPage"
import Breadcrumb from "@/components/common/Breadcrumb"
import huxButton from "@/components/common/huxButton"
import HuxDataTable from "@/components/common/dataTable/HuxDataTable.vue"
import Avatar from "@/components/common/Avatar.vue"
import Size from "@/components/common/huxTable/Size.vue"
import TimeStamp from "@/components/common/huxTable/TimeStamp.vue"
import MenuCell from "@/components/common/huxTable/MenuCell.vue"
import LookAlikeAudience from "./Configuration/Drawers/LookAlikeAudience"
import Icon from "@/components/common/Icon.vue"
import Status from "../../components/common/Status.vue"
import Tooltip from "../../components/common/Tooltip.vue"
import Logo from "../../components/common/Logo.vue"
import HuxAlert from "@/components/common/HuxAlert.vue"

export default {
  name: "Audiences",
  components: {
    PageHeader,
    Breadcrumb,
    huxButton,
    EmptyPage,
    HuxDataTable,
    Avatar,
    Size,
    TimeStamp,
    MenuCell,
    LookAlikeAudience,
    Icon,
    Status,
    Tooltip,
    Logo,
    HuxAlert,
  },
  data() {
    return {
      flashAlert: false,
      alert: {
        type: "success",
        message: "",
      },
      breadcrumbItems: [
        {
          text: "Audiences",
          disabled: true,
          icon: "audiences",
        },
      ],
      columnDefs: [
        {
          text: "Audience name",
          value: "name",
          width: "331px",
          fixed: true,
          divider: true,
          class: "fixed-header",
        },
        {
          text: "Status",
          value: "status",
          width: "160px",
        },
        {
          text: "Size",
          value: "size",
          width: "112px",
        },
        {
          text: "Destinations",
          value: "destinations",
          width: "150px",
        },
        {
          text: "Last delivered",
          value: "last_delivered",
          width: "162",
        },
        {
          text: "Last updated",
          value: "update_time",
          width: "154",
        },
        {
          text: "Last updated by",
          value: "updated_by",
          width: "148",
        },
        {
          text: "Created",
          value: "create_time",
          width: "154",
        },
        {
          text: "Created by",
          value: "created_by",
          width: "100%",
        },
      ],
      loading: false,
      selectedAudience: null,
      showLookAlikeDrawer: false,
    }
  },
  computed: {
    ...mapGetters({
      rowData: "audiences/list",
    }),
    audienceList() {
      let audienceValue = this.rowData
      return audienceValue.sort((a, b) =>
        a.name.toLowerCase() === b.name.toLowerCase()
          ? 0
          : a.name.toLowerCase() < b.name.toLowerCase()
          ? -1
          : 1
      )
    },
    isDataExists() {
      if (this.rowData) return this.rowData.length > 0
      return false
    },
  },
  async mounted() {
    this.loading = true
    await this.getAllAudiences()
    this.loading = false
  },
  methods: {
    ...mapActions({
      getAllAudiences: "audiences/getAll",
    }),

    getActionItems(audience) {
      // This assumes we cannot create a lookalike audience from a lookalike audience
      let isLookalikeableActive =
        audience.lookalikeable === "Active" && !audience.is_lookalike

      let actionItems = [
        { title: "Favorite", isDisabled: true },
        { title: "Export", isDisabled: true },
        { title: "Edit", isDisabled: true },
        { title: "Duplicate", isDisabled: true },
        {
          title: "Create a lookalike",
          isDisabled: !isLookalikeableActive,
          menu: {
            title: "Facebook",
            onClick: () => {
              this.openLookAlikeDrawer(audience)
            },
            icon: "facebook",
          },
        },
        { title: "Delete", isDisabled: true },
      ]

      return actionItems
    },

    openLookAlikeDrawer(audience) {
      this.selectedAudience = audience
      this.showLookAlikeDrawer = true
    },
    onError(message) {
      this.alert.type = "error"
      this.alert.message = message
      this.flashAlert = true
    },
  },
}
</script>
<style lang="scss" scoped>
.audiences-wrap {
  ::v-deep .menu-cell-wrapper .action-icon {
    display: none;
  }
  .page-header--wrap {
    box-shadow: 0px 1px 1px -1px var(--v-black-lighten3),
      0px 1px 1px 0px var(--v-black-lighten3),
      0px 1px 2px 0px var(--v-black-lighten3) !important;
  }
  .top-bar {
    margin-top: 1px;
    .v-icon--disabled {
      color: var(--v-black-lighten3) !important;
      font-size: 24px;
    }
    .text--refresh {
      margin-right: 10px;
    }
  }

  .hux-data-table {
    margin-top: 1px;
    table {
      tr {
        td {
          font-size: 14px;
          height: 63px;
        }
      }
      tbody {
        tr:last-child {
          td {
            border-bottom: 1px solid var(--v-black-lighten3) !important;
          }
        }
      }
    }
  }
  ::v-deep .menu-cell-wrapper :hover .action-icon {
    display: initial;
  }
  .icon-border {
    cursor: default !important;
  }
}
.radio-div {
  margin-top: -11px !important;
}
</style>
