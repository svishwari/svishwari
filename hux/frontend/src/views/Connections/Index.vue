<template>
  <page max-width="100%">
    <div slot="header">
      <page-header>
        <template slot="left">
          <div>
            <breadcrumb :items="breadcrumbs" />
          </div>
          <div class="text-subtitle-1 font-weight-regular">
            Insights into the consumer data that is collected from both online,
            offline, and 3rd party channels.
          </div>
        </template>
      </page-header>

      <page-header header-height="71">
        <template #left>
          <v-btn disabled icon color="black">
            <v-icon medium>mdi-magnify</v-icon>
          </v-btn>
        </template>

        <template #right>
          <huxButton
            button-text="Request a data source"
            variant="primary"
            size="large"
            is-tile
            height="40"
            class="ma-2 font-weight-regular no-shadow mr-0 caption"
            data-e2e="addDataSource"
            @click="toggleDrawer()"
          >
            Request a data source
          </huxButton>
        </template>
      </page-header>

      <v-progress-linear :active="loading" :indeterminate="loading" />
    </div>
    <div v-if="!loading">
      <v-row v-if="isConnectionStarted">
        <v-col>
          <data-sources-list
            @onAddDatasource="toggleDrawer()"
          ></data-sources-list>
        </v-col>
      </v-row>
      <div v-else class="empty-state-wrap text-center">
        <v-icon color="primary lighten-8" x-large>
          mdi-alert-circle-outline
        </v-icon>
        <div class="text-h3">Oops! Thereâ€™s nothing here yet</div>
        <div class="font-weight-regular text-h6 my-2">
          To create a connection, you need to add a destination or a data
          source!
          <br />
          Begin by selecting a button below.
        </div>
        <router-link
          :to="{ name: 'DestinationConfiguration' }"
          class="text-decoration-none"
        >
          <huxButton
            icon="mdi-plus"
            icon-position="left"
            variant="primary"
            size="small"
            icon-size="small"
            :is-tile="true"
            class="ma-2 text-h6 font-weight-regular"
          >
            Destination
          </huxButton>
        </router-link>
        <router-link
          :to="{ name: 'DataSources', params: { select: true } }"
          class="text-decoration-none"
        >
          <huxButton
            button-text="Data source"
            icon="mdi-plus"
            icon-position="left"
            variant="primary"
            size="small"
            :is-tile="true"
            class="ma-2 text-h6 font-weight-regular"
          >
            Data source
          </huxButton>
        </router-link>
      </div>
    </div>
    <data-source-configuration v-model="drawer" />
  </page>
</template>

<script>
import { mapGetters, mapActions } from "vuex"

import DataSourcesList from "./DataSourcesList"
import Page from "@/components/Page"
import PageHeader from "@/components/PageHeader"
import Breadcrumb from "@/components/common/Breadcrumb"
import huxButton from "@/components/common/huxButton"
import DataSourceConfiguration from "@/views/DataSources/Configuration"

export default {
  name: "DataSources",

  components: {
    DataSourcesList,
    Page,
    PageHeader,
    Breadcrumb,
    huxButton,
    DataSourceConfiguration,
  },

  data() {
    return {
      breadcrumbs: [
        {
          text: "Data Sources",
          icon: "data-source",
        },
      ],
      drawer: false,
      loading: false,
    }
  },

  computed: {
    ...mapGetters({
      dataSources: "dataSources/list",
      destinations: "destinations/list",
    }),

    isConnectionStarted() {
      const availableDataSources = this.dataSources.filter(
        (each) => each.is_added
      )
      const availableDestinations = this.destinations.filter(
        (each) => each.is_added
      )
      return availableDataSources.length > 0 || availableDestinations.length > 0
    },
  },

  watch: {
    $route() {
      if (this.$route.params.select) {
        this.drawer = true
      } else {
        this.drawer = false
      }
    },
  },

  async mounted() {
    this.loading = true
    await this.getDataSources()
    await this.getDestinations()
    this.loading = false

    if (this.$route.params.select) {
      this.drawer = true
    }

    this.$root.$on("same-route-Connections", () => {
      this.toggleDrawer()
    })
  },

  methods: {
    ...mapActions({
      getDataSources: "dataSources/getAll",
      getDestinations: "destinations/getAll",
    }),
    toggleDrawer() {
      this.drawer = !this.drawer
    },
  },
}
</script>

<style lang="scss" scoped>
.empty-state-wrap {
  padding-top: 160px;
}
</style>
