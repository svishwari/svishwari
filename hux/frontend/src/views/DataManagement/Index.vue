<template>
  <div class="datasource-wrap">
    <page-header :header-height-changes="'py-3'">
      <template slot="left">
        <breadcrumb :items="breadcrumbItems" />
      </template>
    </page-header>
  </div>
</template>

<script>
import PageHeader from "@/components/PageHeader"
import Breadcrumb from "@/components/common/Breadcrumb"
export default {
  name: "DataSources",
  components: { PageHeader, Breadcrumb },
  data() {
    return {
      breadcrumbItems: [
        {
          text: "Data Sources",
          disabled: true,
          icon: "datasource",
        },
      ],
    }
  },
}
</script>

<style lang="scss" scoped></style>
