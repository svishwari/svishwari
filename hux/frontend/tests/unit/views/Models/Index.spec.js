//File to be tested
import Index from "@/views/Models/Index.vue"
//Test utils
import { shallowMount, createLocalVue } from "@vue/test-utils"
//Dependencies
import Store from "@/store/index.js"
import Vuex from "vuex"
import Vuetify from "vuetify"
import filters from "@/filters"
//Miragejs
import { makeServer } from "@/api/mock/server.js"
import modelSeeds from "@/api/mock/seeds/model.js"

let server

let vuetify

beforeEach(() => {
  vuetify = new Vuetify()
  server = makeServer({ environment: "test" })
  modelSeeds.forEach((seed) => server.create("model", seed))
})

afterEach(() => {
  server.shutdown()
})

const localVue = createLocalVue()
localVue.use(Vuex)
const store = Store

Object.keys(filters).forEach((filterName) => {
  localVue.filter(filterName, filters[filterName])
})

describe("Listing of models", () => {
  test("Models store successfully filled", async function () {
    const wrapper = shallowMount(Index, {
      store,
      vuetify,
      localVue,
      sync: false,
    })
    // TODO: find a better way to this
    // Here we are manually triggering the api call
    await wrapper.vm.getModels()
    expect(store.getters["models/list"].length).toBe(wrapper.vm.models.length)
    expect(wrapper.find("descriptive-card-stub").exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })
})
