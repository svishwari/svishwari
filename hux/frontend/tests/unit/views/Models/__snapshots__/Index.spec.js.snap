// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing of models Models store successfully filled 1`] = `
<div>
  <page-header-stub bgcolor="white" headerheight="110" headerpadding="px-8" headerheightchanges="py-5" data-e2e="models-header"></page-header-stub>
  <v-progress-linear-stub color="primary" value="0" buffervalue="100" height="4"></v-progress-linear-stub>
  <v-row-stub tag="div" data-e2e="models-list" class="pa-14">
    <descriptive-card-stub icon="model-ltv" title="Predicted lifetime value" description="Predicts the lifetime value of a customer based on models." height="255" width="280" dotoption="Activate" iconcolor="Primary" data-e2e="model-item"><template></template> <template></template> <template></template></descriptive-card-stub>
    <descriptive-card-stub icon="model-purchase" title="Propensity to purchase" description="Propensity of a customer making a purchase after receivin..." height="255" width="280" dotoption="Activate" iconcolor="Primary" data-e2e="model-item"><template></template> <template></template> <template></template></descriptive-card-stub>
    <descriptive-card-stub icon="model-unsubscribe" title="Propensity to unsubscribe" description="Propensity of a customer unsubscribing to emails." height="255" width="280" dotoption="Activate" iconcolor="Primary" data-e2e="model-item"><template></template> <template></template> <template></template></descriptive-card-stub>
    <descriptive-card-stub icon="model-unsubscribe" title="Churn" description="Rating forecast for change in customer numbers." disabled="true" actionmenu="true" height="255" width="280" dotoption="Activate" iconcolor="Primary" data-e2e="model-item"><template></template>
      <!----> <template></template>
    </descriptive-card-stub>
  </v-row-stub>
</div>
`;
